dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -180,9 +180,13 @@ def login
 
     prepare_locale_cookie user
 
-    if allows_silent_takeover(user, auth_hash)
-      user = silent_takeover user, auth_hash
-      sign_in_user user
+    if email_already_taken(user)
```

Comment:
won't this new conditional break the one below it?

https://github.com/code-dot-org/code-dot-org/blob/e921188d49ed1147b86affb51d7e568a8a20cf0b/dashboard/app/controllers/omniauth_callbacks_controller.rb#L193-L197